sdv/metadata/single_table.py

```diff
@@ -250,55 +248,90 @@ def _validate_datatype(id):
         """Check whether id is a string or a tuple of strings."""
         return isinstance(id, str) or isinstance(id, tuple) and all(isinstance(i, str) for i in id)
 
+    def _validate_key(self, id, key_type):
```

Comment:
The previous implementation had the validation inside the set_key methods. Pulling it out of the key methods so I can use it in the `validate` method.